### DISCUSSION TITLE
Client side file encryption - beginner question

### DISCUSSION THREAD

**Discussion by kalmarz** [reflex-dev/reflex#3851] [Q&A] [answered]
I'm wondering what's the "reflex way" of implementing a client-side file encryption, without uploading anything to the backend? I want the user to pick a file from their computer, the app would generate a key, encrypt the file and save it locally.

**Comment by picklelo** [comment, marked as answer]
Any logic written in the state will happen on the backend by default. If you want to have client side logic, you can use `rx.call_script` to run Javascript on the browser. See [here](https://reflex.dev/docs/api-reference/browser-javascript/#inline-scripts) for more (this is our backdoor to use plain JS within Reflex)

**Reply by kalmarz** [reply]
Thanks for the explanation, this will do.
